fast/stages/2-networking-a-peering/main.tf

```diff
@@ -17,13 +17,20 @@
 # tfdoc:file:description Networking folder and hierarchical policy.
 
 locals {
+  _region_tokens = { for v in local._regions : v => split("-", v) }
+  _regions = distinct(concat(
+    [var.regions.primary, var.regions.secondary],
+    values(module.dev-spoke-vpc.subnet_regions),
+    values(module.landing-vpc.subnet_regions),
+    values(module.prod-spoke-vpc.subnet_regions),
+  ))
   custom_roles = coalesce(var.custom_roles, {})
-  l7ilb_subnets = {
-    for env, v in var.l7ilb_subnets : env => [
-      for s in v : merge(s, {
-        active = true
-        name   = "${env}-l7ilb-${s.region}"
-    })]
+  region_shortnames = {
```

Comment:
This would misbehave in case of regions with a double digit (e.g. `europe-westXX`)

Reply:
maybe
```hcl
region_shortnames = {
  for v in local._regions : 
  v => join("", regex("([a-z])[a-z]+-([a-z])[a-z]+(\\d+)", "europe-west1"))
}
```

But that will still not work with something like asia-south1 and asia-southest1

Reply:
this is what the current implementation spits out, I will update as per Simo's point and share the updated results

```hcl
tmp = {
  "asia-east1" = "ae1"
  "asia-east2" = "ae2"
  "asia-northeast1" = "an1"
  "asia-northeast2" = "an2"
  "asia-northeast3" = "an3"
  "asia-south1" = "as1"
  "asia-south2" = "as2"
  "asia-southeast1" = "as1"
  "asia-southeast2" = "as2"
  "australia-southeast1" = "as1"
  "australia-southeast2" = "as2"
  "europe-central2" = "ec2"
  "europe-north1" = "en1"
  "europe-southwest1" = "es1"
  "europe-west1" = "ew1"
  "europe-west2" = "ew2"
  "europe-west3" = "ew3"
  "europe-west4" = "ew4"
  "europe-west6" = "ew6"
  "europe-west8" = "ew8"
  "europe-west9" = "ew9"
  "me-west1" = "mw1"
  "northamerica-northeast1" = "nn1"
  "northamerica-northeast2" = "nn2"
  "southamerica-east1" = "se1"
  "southamerica-west1" = "sw1"
  "us-central1" = "uc1"
  "us-east1" = "ue1"
  "us-east4" = "ue4"
  "us-east5" = "ue5"
  "us-south1" = "us1"
  "us-west1" = "uw1"
  "us-west2" = "uw2"
  "us-west3" = "uw3"
  "us-west4" = "uw4"
}
```

Reply:
so, slight tweak and I think it works

```hcl
  region_shortnames = {
    for k, v in local._regions : k => join("", [
      substr(v.0, 0, 1),
      substr(v.1, 0, 1),
      regex("[0-9]+$", v.1),
    ])
  }
  _regions = { for v in local.regions : v => split("-", v) }
  ```
  
result

```hcl
tmp = {
  "asia-east1" = "ae1"
  "asia-east2" = "ae2"
  "asia-northeast1" = "an1"
  "asia-northeast2" = "an2"
  "asia-northeast3" = "an3"
  "asia-south1" = "as1"
  "asia-south2" = "as2"
  "asia-southeast1" = "as1"
  "asia-southeast2" = "as2"
  "australia-southeast1" = "as1"
  "australia-southeast2" = "as2"
  "europe-central2" = "ec2"
  "europe-north1" = "en1"
  "europe-southwest1" = "es1"
  "europe-west1" = "ew1"
  "europe-west12" = "ew12"
  "europe-west2" = "ew2"
  "europe-west3" = "ew3"
  "europe-west4" = "ew4"
  "europe-west6" = "ew6"
  "europe-west8" = "ew8"
  "europe-west9" = "ew9"
  "me-west1" = "mw1"
  "northamerica-northeast1" = "nn1"
  "northamerica-northeast2" = "nn2"
  "southamerica-east1" = "se1"
  "southamerica-west1" = "sw1"
  "us-central1" = "uc1"
  "us-east1" = "ue1"
  "us-east4" = "ue4"
  "us-east5" = "ue5"
  "us-south1" = "us1"
  "us-west1" = "uw1"
  "us-west2" = "uw2"
  "us-west3" = "uw3"
  "us-west4" = "uw4"
}
```

Reply:
ouch, asia southeast